### PR TITLE
Update nightly-build workflow

### LINKED ISSUE #419: Request to push nightly images to a separate repo
As you can see from the following screenshot. There are too many image tags. It's hard to find useful tags.

So, I suggest that we push the nightly images to a separate repo. For instance, we can push `ghcr.io/kubesphere/devops-tools:nightly-20211221` to `ghcr.io/kubesphere-nightly/devops-tools:nightly-20211221`.

image

## References
* https://github.com/kubesphere/ks-devops/blob/master/.github/workflows/nightly-build.yaml

/good-first-issue

### PR DESCRIPTION
**Pull request by chengleqi**
### What type of PR is this?

/kind chore

### What this PR does / why we need it:
- Change the destination of nightly-build container images from `ghcr.io/kubesphere` to `ghcr.io/kubesphere-nightly`.
- Remain the `ghcr.io/${{ github.repository_owner }}` destination registry for contributors.

### Which issue(s) this PR fixes:

Fixes #419 

### Special notes for reviewers:
Should create **a new repo** and add the **`GHCR_TOKEN_NIGHTLY`** secret.

Please check the following list before waiting reviewers:

- [ ] Already committed the CRD files to [the Helm Chart](https://github.com/kubesphere-sigs/ks-devops-helm-chart/) if you created some new CRDs
- [ ] Already [added the permission](https://github.com/kubesphere/ks-installer/blob/9e063b085a0e43fdb3d0d9e3e7f4149146f14b9c/roles/ks-core/prepare/files/ks-init/role-templates.yaml) for the new API
- [ ] Already added the RBAC markers for the new controllers

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
I have tried to push a nightly image to `kubesphere-nightly`. But I got an error. See also:

```yaml
[root@node1 ~]# docker tag  ghcr.io/kubesphere/devops-apiserver:nightly-20220815 ghcr.io/kubesphere-nightly/devops-apiserver:nightly-20220815
[root@node1 ~]# docker push ghcr.io/kubesphere-nightly/devops-apiserver:nightly-20220815
The push refers to repository [ghcr.io/kubesphere-nightly/devops-apiserver]
54e1cb801a21: Mounted from kubesphere/devops-apiserver
9d857fec6b6c: Mounted from kubesphere/devops-apiserver
8d7366c22fd8: Mounted from kubesphere/devops-apiserver
denied: not_found: owner not found
```

So, I'm wondering if we could use `kubesphere-sigs` instead.

**Comment by chengleqi** [comment]
Probably because the user or organization named `kubesphere-nightly` doesn't exist.
I'll change the `kubesphere-nightly` to `kubesphere-sigs`.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

hi @lxm , please feel free to help to review this. Thanks!

**Comment by lxm** [comment]
lgtm for this pr, but I think the configuration about the `registry` and `repo` should be written into the document

**Comment by chengleqi** [comment]
And it would be nicer to remind the contributors who fork this project to set `GHCR_TOKEN` action secret.
Otherwise the nightly-build workflow will failed and contributors will receive a run failed notification email  from GitHub every night.

File: .github/workflows/nightly-build.yaml [hunk 1]
```diff
@@ -160,13 +174,20 @@ jobs:
...
           username: ${{ secrets.DOCKER_HUB_USER }}
           password: ${{ secrets.DOCKER_HUB_SECRETS }}
       - name: Login to GHCR
-        if: github.event_name != 'pull_request'
+        if: github.event_name != 'pull_request' && github.repository_owner == 'kubesphere'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: kubesphere-sigs
+          password: ${{ secrets.GHCR_TOKEN_NIGHTLY }}
```
Comment (LinuxSuRen) [inline .github/workflows/nightly-build.yaml:188]: We could still use the following token.

`GHCR_TOKEN`

**Comment by LinuxSuRen** [comment]
> And it would be nicer to remind the contributors who fork this project to set `GHCR_TOKEN` action secret. Otherwise the nightly-build workflow will failed and contributors will receive a run failed notification email from GitHub every night.

Is that possible not to login if the repository_owner is not `kubesphere` and the token is empty.

**Comment by chengleqi** [comment]
### kubesphere/ks-devops repo
- push to `kubespheredev/devops-apiserver:nightly-xxxxxxx` and `ghcr.io/kubesphere-sigs/devops-apiserver:nightly-xxxxxxx`.

### user/ks-devops repo
- set `GHCR_TOKEN`: push to `user/devops-apiserver:nightly-xxxxxxx`
- **NOT** set `GHCR_TOKEN`: skip the NightlyBuild jobs by check job.

File: .github/workflows/nightly-build.yaml [hunk 2]
```diff
@@ -38,20 +56,20 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
       - name: Login to DockerHub
-        if: github.event_name != 'pull_request' && github.repository_owner == 'kubesphere'
+        if: github.repository_owner == 'kubesphere'
```
Comment (LinuxSuRen) [inline .github/workflows/nightly-build.yaml:59]: GitHub Actions do not allow to read the `secrets` if it's a PR. So, I guess that we still need the following condition.

`github.event_name != 'pull_request'`

File: .github/workflows/nightly-build.yaml [hunk 3]
```diff
@@ -63,20 +81,23 @@ jobs:
         with:
           file: config/dockerfiles/controller-manager/Dockerfile
           tags: ${{ steps.meta.outputs.tags }}
-          push: ${{ github.event_name != 'pull_request' }}
+          push: true
```
Comment (LinuxSuRen) [inline .github/workflows/nightly-build.yaml:84]: It looks like we cannot push the image if this is a PR due to we cannot using the secret.

File: .github/workflows/nightly-build.yaml (hunk 2 above)
Comment (chengleqi) [reply .github/workflows/nightly-build.yaml:59]: I think this NightlyBuild workflow only trigger by cron (schedule type event).
PR won't trigger this workflow.

Comment (LinuxSuRen) [reply .github/workflows/nightly-build.yaml:59]: Fair enough.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Pull request merged**